sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTimeWindows.scala

```diff
@@ -49,11 +49,23 @@ object TimeWindowing extends Rule[LogicalPlan] {
    * The windows are calculated as below:
    * maxNumOverlapping <- ceil(windowDuration / slideDuration)
    * for (i <- 0 until maxNumOverlapping)
-   *   lastStart <- timestamp - (timestamp - startTime + slideDuration) % slideDuration
-   *   windowStart <- lastStart - i * slideDuration
+   *   lastStart <- timestamp - (timestamp - startTime) % slideDuration
+   *   lastStartAdjusted <- (timestamp < startTime) ? lastStart - slideDuration : lastStart
+   *   windowStart <- lastStartAdjusted - i * slideDuration
    *   windowEnd <- windowStart + windowDuration
    *   return windowStart, windowEnd
    *
+   * Rationale of lastStartAdjusted:
```

Comment:
This doesn't belong in user-facing javadoc IMHO

Reply:
May I know what's the suggestion here? Should I move this explanation to somewhere else?

Reply:
You could put it inside the function as comments, or omit it

Reply:
I put it in the description of this PR